### DISCUSSION TITLE
Using Payload and Perkeep Together?

### DISCUSSION THREAD

**Discussion by WDervish** [payloadcms/payload#1135] [Q&A]
Hello everybody, I am new to software development. I am a civil engineer and building a software solution for the industry. I was told to look into Payload and Perkeep. I have been doing my research. I am not a programmer, but trying to learn so that I can communicate with the developers.

I would like to clarify few things and ask few questions. My understanding is that Payload is a headless CMS tool for content management. Perkeep is a software to manage content; index, search, sync and share data. Documents in Perkeep has a metadata, which is used for querying etc. 

Are any of you familiar with Perkeep? I am trying to understand the difference between the two and how they can work together. Can one integrate Payload and Perkeep together? and if so, would one usecase be, write a code with Payload to define how the content will be managed, integrate it with Perkeep, and have Perkeep index, search, sync and share; basically using both together? What are your thoughts, how can these two be used together?

Regards,

Omer

**Comment by DanRibbens** [comment]
What problem are you trying to solve by introducing Perkeep into your tech stack? I'm trying to understand what it is going to offer beyond what you already get with Payload.

If you want to have a file storage, I recommend the [plugin-cloud-storage](https://github.com/payloadcms/plugin-cloud-storage) to put files on your cloud host, separate from your node server.

Other integrations we see with Payload are writing hooks that populate a search solution like Algolia. That is a very typical pattern when querying requirements go beyond just finding and filtering some data.

Maybe there is a use-case where it makes sense to integrate the two, but I don't know what your project scope and constraints are to say if that is a good architecture or not. Ultimately you want your developer involved in these decisions. 

Does this help at all?

**Reply by WDervish** [reply]
Dan, I appreciate the input. Thank you! I am thinking Perkeep for version control of various files. Can you do version control of files with Payload? Have each files contain a metadata then, provide the files to users based on their search parameters?

Thank you.
